www/class/centreonLDAP.class.php

```diff
@@ -968,21 +968,29 @@ public function isSyncNeededAtLogin(int $arId = null, int $contactId): bool
 
                 // checking if the interval between two synchronizations is reached
                 $currentTime = time();
+
                 if (
-                    ($syncState['ldap_sync_interval'] * 3600 + $ldapBaseSync['referenceDate']) <= $currentTime
-                    && $manualOverride['contact_ldap_last_sync'] < $ldapBaseSync['referenceDate']
+                    ($syncState['ldap_sync_interval'] * 3600 + $contactData['contact_ldap_last_sync']) <= $currentTime
+                    && $contactData['contact_ldap_last_sync'] < $ldapBaseSync['referenceDate']
                 ) {
                     // synchronization is expected
                     $this->centreonLog->insertLog(
                         3,
-                        'LDAP AUTH : Updating user DN of ' . $manualOverride['contact_name']
+                        'LDAP AUTH : Updating user DN of ' . $contactData['contact_name']
                     );
                     return true;
                 }
             }
         } catch (\PDOException $e) {
-            throw new \Exception('Error while getting automatic synchronization value for LDAP Id : ' . $arId);
+            $this->centreonLog->insertLog(
+                3,
+                'Error while getting automatic synchronization value for LDAP Id : ' . $arId
+            );
             // assuming it needs to be synchronized
+            $this->centreonLog->insertLog(
+                3,
+                'LDAP AUTH : Updating user DN of ' . $contactData['contact_name']
```

Comment:
last thing, $contactData is not necessarily defined
```suggestion
                'LDAP AUTH : Updating user DN of ' .
                (!empty($contactData['contact_name']) ? $contactData['contact_name'] : "contact id $contactId")
```